test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -151,9 +152,34 @@ private void wipeCluster() throws IOException {
         // wipe index templates
         adminClient().performRequest("DELETE", "_template/*");
 
-        // wipe snapshots
-        // Technically this deletes all repositories and leave the snapshots in the repository. OK.
-        adminClient().performRequest("DELETE", "_snapshot/*");
+        wipeSnapshots();
+    }
+
+    /**
+     * Wipe fs snapshots we created one by one and all repositories so that the next test can create the repositories fresh and they'll
+     * start empty. There isn't an API to delete all snapshots. There is an API to delete all snapshot repositories but that leaves all of
+     * the snapshots intact in the repository.
+     */
+    private void wipeSnapshots() throws IOException {
+        for (Map.Entry<String, ?> repo : entityAsMap(adminClient.performRequest("GET", "_snapshot/_all")).entrySet()) {
+            String repoName = repo.getKey();
+            Map<?, ?> repoSpec = (Map<?, ?>) repo.getValue();
+            String repoType = (String) repoSpec.get("type");
+            if (repoType.equals("fs")) {
+                // All other repo types we really don't have a chance of being able to iterate properly, sadly.
+                String url = "_snapshot/" + repoName + "/_all";
+                Map<String, String> params = singletonMap("ignore_unavailable", "true");
+                List<?> snapshots = (List<?>) entityAsMap(adminClient.performRequest("GET", url, params)).get("snapshots");
+                for (Object snapshot : snapshots) {
+                    Map<?, ?> snapshotInfo = (Map<?, ?>) snapshot;
+                    String name = (String) snapshotInfo.get("snapshot");
+                    logger.debug("wiping snapshot [{}/{}]", repoName, name);
+                    adminClient().performRequest("DELETE", "_snapshot/" + repoName + "/" + name);
+                }
+            }
+            logger.debug("wiping snapshot repository [{}]", repoName);
+            adminClient().performRequest("DELETE", "_snapshot/" + repoName);
+        }
```

Comment:
thanks @nik9000 ! @elastic/es-clients is this ok? I guess all the client runners will have to be changed accordingly.